spec/services/login_adoption_service_spec.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe LoginAdoptionService do
+  let(:service) { LoginAdoptionService.new(user) }
+
+  context 'organic modal experiment' do
```

Comment:
I'd organize these specs differently:

`describe` at top level, which is simply the name of the function:
`describe #perform`

`context` within the `describe` block, which are described as conditionals:
`context 'when organic modal flag is off'`
`context 'when organic modal flag is on'`

`it` within the `context` blocks, which describe the results:
`it returns a hash with false organic_modal key`
`it returns a hash with true organic_modal key`

For a unit spec, you should be able to just follow the logic of your code, and split out `contexts` whenever you see a conditional

Reply:
This is the conditional flow of your service code, which is how your contexts should be organized:

- When Flipper is enabled
  - When user is dslogon authenticated
    - When user has ivc
      - hash returns false 
    - When user has avc
      - hash returns false
    - When user has no ivc/avc
      - hash returns true  
  - When user is not dslogon authenticated
    - hash returns false 
- When Flipper is disabled
  - hash returns false